oop.md

```diff
@@ -0,0 +1,179 @@
+# Singleton
```

Comment:
Yes, that's a good example of a Singleton design patter. It ensures that there is only one instance of a class and provides a global point of access to that instance. In this case, the `SingletonInstance` class creates and returns a single instance of the `AxiosInstance` class, which can be accessed globally through the `getInstance` method. The private constructor ensures that no new instances of `SingletonInstance` can be created from outside the class. GJ!